src/ansys/pre_commit_hooks/add_license_headers.py

```diff
@@ -183,17 +190,13 @@ def list_noncompliant_files(args: argparse.Namespace, proj: project.Project) ->
         List of the files that are missing license headers.
     """
     # Create a temporary file containing lint.run json output
-    filename = None
-    with NamedTemporaryFile(mode="w", delete=False) as tmp:
-        args.json = True
-        lint.run(args, proj, tmp)
-        filename = tmp.name
-
-    # Open the temporary file, load the JSON file, and find files that
-    # are missing license headers.
-    lint_json = None
-    with open(filename, "rb") as file:
-        lint_json = json.load(file)
+    from io import StringIO
+
+    buffer = StringIO()
+    args.json = True  # Output as json
+    lint.run(args, proj, buffer)
+    lint_json = json.loads(buffer.getvalue())
+    buffer.close()
```

Comment:
This is a nice approach, but please move the import statement to its proper location... at the beginning of the module

Reply:
Well.. too bad this PR is closed. 🤣

Reply:
You can always open a new one =)